libr/anal/esil2reil.c

```diff
@@ -0,0 +1,1124 @@
+/*
+ * Convert from ESIL to REIL (Reverse Engineering Intermediate Language)
+ * Contributor: sushant94
+ */
+
+#include <r_anal.h>
+
+#define REIL_TEMP_PREFIX "V"
+void reil_flag_spew_inst(RAnalEsil *esil, const char *flag);
+const char *ops[] = { FOREACHOP(REIL_OP_STRING) };
+
+// Get size of a register.
+static ut8 esil_internal_sizeof_reg(RAnalEsil *esil, const char *r) {
+	RRegItem *i;
+	if (!esil || !esil->anal || !esil->anal->reg || !r)
+		return R_FALSE;
+	i = r_reg_get(esil->anal->reg, r, -1);
+	if (!i)
+		return R_FALSE;
+	return (ut8)i->size;
+}
+
+RAnalReilArgType reil_get_arg_type(RAnalEsil *esil, char *s) {
+	if (!strncmp(s, REIL_TEMP_PREFIX, strlen(REIL_TEMP_PREFIX)))
+		return ARG_TEMP;
+
+	int type = r_anal_esil_get_parm_type(esil, s);
+
+	switch (type) {
+		case R_ANAL_ESIL_PARM_REG:
+			return ARG_REG;
+		case R_ANAL_ESIL_PARM_NUM:
+			return ARG_CONST;
+		case R_ANAL_ESIL_PARM_INTERNAL:
+			return ARG_ESIL_INTERNAL;
+		default:
+			return ARG_NONE;
+	}
+}
+
+// Marshall the struct into a string
+void reil_push_arg(RAnalEsil *esil, RAnalReilArg *op) {
+	char tmp_buf[32];
+	snprintf(tmp_buf, sizeof(tmp_buf), "%s:%d", op->name, op->size);
+	r_anal_esil_push(esil, tmp_buf);
+}
+
+// Unmarshall the string in stack to the struct.
+RAnalReilArg *reil_pop_arg(RAnalEsil *esil) {
+	RAnalReilArg *op;
+	int i, j = 0, flag = 0, len;
+	char tmp_buf[32];
+	char *buf = r_anal_esil_pop(esil);
+	if (!buf) return NULL;
+
+	len = strlen(buf);
+	op = R_NEW0(RAnalReilArg);
+
+	for (i = 0; i < len; i++) {
+		if (buf[i] == ':') {
+			tmp_buf[j] = '\0';
+			strncpy(op->name, tmp_buf, sizeof(op->name));
+			memset(tmp_buf, 0, sizeof(tmp_buf));
+			j = 0;
+			flag = 1;
+			continue;
+		}
+		// Strip all spaces
+		if (buf[i] == ' ') continue;
+		tmp_buf[j] = buf[i];
+		j++;
+	}
+
+	tmp_buf[j] = '\0';
+
+	// If we have not encountered a ':' we don't know the size yet.
+	if (!flag) {
+		strncpy(op->name, tmp_buf, sizeof(op->name));
+		op->type = reil_get_arg_type(esil, op->name);
+		if (op->type == ARG_REG) 
+			op->size = esil_internal_sizeof_reg(esil, op->name);
+		else if (op->type == ARG_CONST)
+			op->size = esil->anal->bits;
+		return op;
+	}
+
+	op->size = atoi(tmp_buf);
```

Comment:
Shall we use `op->size =  strtol(tmp_buf, NULL, 10);` instead?
See `man 3 atoi` for details.